core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -161,19 +162,22 @@ final class KafkaMetadataLog private (
 
   override def truncateToLatestSnapshot(): Boolean = {
     val latestEpoch = log.latestEpoch.getOrElse(0)
-    latestSnapshotId().asScala match {
+    val (truncated, forgottenSnapshots) = latestSnapshotId().asScala match {
       case Some(snapshotId) if (snapshotId.epoch > latestEpoch ||
         (snapshotId.epoch == latestEpoch && snapshotId.offset > endOffset().offset)) =>
```

Comment:
unrelated from your change but this indentation threw me off here.

Reply:
I fixed the implementation. Let me know if it is better.